map/src/util/Utils.js

```diff
@@ -79,9 +79,16 @@ function getProfileIcon(profile, color) {
     }
 }
 
-export function mergeStateObject(getObject, setFunction, todoObject) {
-    const merged = Object.assign({}, getObject, todoObject);
-    setFunction(() => merged); // is really need => ?
+export function mergeStateObject(getObject, setFunction, todoObject, path = null) {
+    /*
+        Please note: getObject will be directly modified here!
+        In the next render cycle it will be overriden by useState.
+        We use ()=>{} to make correct queue of changes for next render.
+
+        You might use return Object as you want, but you don't have to re-assign it to getObject (as it's already done).
+    */
+    const merged = Object.assign(getObject, todoObject); // getObject has been modified
```

Comment:
delete comments

Reply:
This comments are really important and might be used as documentation: how this function works in context of React state values concept.